File: php/birthday.php
<?php

  $months = array(
   1 => "January",
   2 => "February",
   4 => "April",
   5 => "May",
   6 => "June",
   8 => "August",
   9 => "September",
   10 => "October",
   11 => "November",
   12 => "December");
?>

<html>
  <head>
    <title>Birthday Program</title>
  </head>

  <body>
    
    <h1>Birthday Calculator Program</h1>

    <p>Greetings.  Please enter your name and your birtday below, then hit
      "submit".</p>
      <form method="post" action="birthday_result.php">
      <table>
        <tr>
          <td>Name:</td>
          <td><input type='text' name='name'/></td>
        </tr>
        <tr>
          <td>Birthday:</td>
          <td>
           <select name="month">
	     <?php 
		foreach($months as $key => $month) {
		  print "<option value='" . $key . "'>" . $month . "</option>\n";
		}
              ?>
            </select>
	    <select name="date">
	      <?php
	        //TODO: create a loop to output the days of the month 1..31:

              ?>
	    </select>
            <select name="year">
             <?php
                for($i=1950; $i<=2000; $i++) {
                  print "<option value='" . $i . "'>" . $i . "</option>\n";
                }
              ?>
            </select>
          </td>
        </tr>
        <tr> 
          <td><input type='submit' name='submit' value='Submit'/></td>
        </tr>
      </table>
      </form>
  </body>
</html>

File: python/birthday.py
"""
Interactively prompts the user for their bithdate
and computes how old they are.
"""
import sys
import math
from datetime import date

# A "dictionary" that maps months (as strings) to their ordinal (integers)
monthMap = {
    "January": 1,
    "February": 2,
    "March": 3,
    "April": 4,
    "May": 5,
    "June": 6,
    "July": 7,
    "August": 8,
    "September": 9,
    "October": 10,
    "November": 11,
    "December": 12
}

# An alternative way of creating the dictionary/map using a list
months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']
# and a dictionary comprehension:
monthMap = { month:months.index(month)+1 for month in months }

today = date.today()


year = input("Enter your birth year (ex: 2022): ")
try:
  year = int(year)
except:
  print("ERROR: invalid input")
  exit(1)

month = input("Enter your birth month (ex: July): ")
try:
  month = monthMap[month]
except:
  print("ERROR: invalid input")
  exit(1)

day = input("Enter your birth day (of month): ")
try:
  day = int(day)
except:
  print("ERROR: invalid input")
  exit(1)

try:
  birthdate = date(year, month, day)
except:
  print("ERROR: invalid input")
  exit(1)

diff = today - birthdate
years = math.floor(diff.days / 365.25)
remain = diff.days - 365.25 * years
weeks = int(remain // 7)
remain = remain - 7 * weeks
days = int(remain)

print(f"You are {years} years, {weeks} weeks and {days} days old")


